### PR TITLE
LCORE-1853: Add relevance cutoff score to inline BYOK RAG

### PR DESCRIPTION
**Pull request by syedriko**
## Description

In the Lightspeed Stack configuration, byok_rag section, added a "relevance_cutoff_score" configuraiton variable, a positive float.
When a RAG chunk is retrieved from a BYOK database and its relevance score is less than the cutoff, the chunk is dropped from further consideration. Specifically, this cutoff filtering happens before score multipliers are applied.
Only added handling of relevance_cutoff_score to the inline BYOK RAG for now as adding it to the tool BYOK RAG is calling for changes in Llama Stack.

## Type of change

- [ ] Refactor
- [x] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up service version
- [ ] Bump-up dependent library
- [ ] Bump-up library or tool used for development (does not change the final image)
- [ ] CI configuration change
- [ ] Konflux configuration change
- [ ] Unit tests improvement
- [ ] Integration tests improvement
- [ ] End to end tests improvement
- [ ] Benchmarks improvement

## Tools used to create PR

Identify any AI code assistants used in this PR (for transparency and review context)

- Assisted-by: Cursor
- Generated by: (e.g., tool name and version; N/A if not used)

## Related Tickets & Documents

- Related Issue # 
- Closes # https://redhat.atlassian.net/browse/LCORE-1853

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [ ] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Please provide detailed steps to perform tests related to this code change.
- How were the fix/results from this change verified? Please provide relevant screenshots or results.

## Summary by CodeRabbit

* **New Features**
  * Per-knowledge-source relevance_cutoff_score for BYOK Inline RAG to filter results by minimum similarity (default 0.3); Tool RAG/file_search unaffected.

* **Documentation**
  * BYOK guide updated with usage, tuning guidance, and Inline vs Tool RAG notes.

* **API**
  * Public schema exposes relevance_cutoff_score with default and validation.

* **Tests**
  * Unit tests expanded for configuration, validation, propagation, and filtering behavior.

### CODE REVIEW

File: docs/devel_doc/openapi.json [hunk 1]
```diff
@@ -11806,6 +11806,13 @@
                         "title": "Score multiplier",
                         "description": "Multiplier applied to relevance scores from this vector store. Used to weight results when querying multiple knowledge sources. Values > 1 boost this store's results; values < 1 reduce them.",
                         "default": 1.0
+                    },
+                    "relevance_cutoff_score": {
+                        "type": "number",
+                        "exclusiveMinimum": 0.0,
+                        "title": "Relevance cutoff score",
+                        "description": "Minimum raw similarity score to consider a result relevant. Results with a similarity score below this threshold are not returned.",
+                        "default": 0.3
```
[the remark this answers is not in the document]
Comment (syedriko) [reply docs/devel_doc/openapi.json]: @coderabbit, which CI pipeline is failing?

[the remark this answers is not in the document]
Comment (syedriko) [reply docs/devel_doc/openapi.json]: @coderabbit, you're saying "The CI pipeline is failing because docs/openapi.json is out of sync with the generated schema. "

**Review by tisnik — APPROVED** [review]
LGTM

**Comment by are-ces** [comment]
Related to #1525

File: src/utils/vector_search.py [hunk 2]
```diff
@@ -410,6 +431,7 @@ async def _fetch_byok_rag(
                     vector_store_id,
                     query,
                     score_multiplier_mapping.get(vector_store_id, 1.0),
+                    _relevance_cutoff_for_vector_store(vector_store_id),
```
Comment (are-ces) [inline src/utils/vector_search.py]: Tiny nit from SWE perspective. Just wondering: since score multiplier and relevance score have the same data structure, would it make sense to retrieve cutoff score the same way?

File: tests/unit/models/config/test_byok_rag.py [hunk 3]
```diff
@@ -199,3 +203,18 @@ def test_byok_rag_configuration_score_multiplier_must_be_positive() -> None:
             db_path="tests/configuration/rag.txt",
             score_multiplier=0.0,
         )
+
+
+def test_byok_rag_configuration_relevance_cutoff_must_be_positive() -> None:
```
Comment (are-ces) [inline tests/unit/models/config/test_byok_rag.py]: Nit, do we also want to test behavior if it is negative?

**Review by are-ces — CHANGES_REQUESTED** [review]
LGTM overall TYVM!

I have a couple of nits which you can ignore but I would like to see some integration test for this functionality

**Comment by syedriko** [comment]
@are-ces PTAL

**Review by are-ces — APPROVED** [review]
LGTM TY

File: docs/user_doc/byok_guide.md [hunk 4]
```diff
@@ -118,6 +120,13 @@ flowchart TD
     Cut --> Context["Final Inline RAG Context"]
 ```
 
+
+> [!NOTE]
+> `relevance_cutoff_score` applies to Inline RAG only. When the model uses Tool RAG (`file_search`),
+> Lightspeed Stack does not send this setting; retrieval uses Llama Stack’s default ranking for that path.
```
Comment (are-ces) [inline docs/user_doc/byok_guide.md]: ogx

File: tests/unit/utils/test_vector_search.py [hunk 5]
```diff
@@ -27,10 +29,37 @@
...
 )
 
 
+def _vector_io_query_stub_like_backend(
+    chunk_score_pairs: list[tuple[Any, float]], mocker: MockerFixture
+) -> Callable[..., Awaitable[Any]]:
+    """Build an async ``vector_io.query`` stand-in that honors ``score_threshold``.
+
+    Production code forwards ``relevance_cutoff_mapping`` values as ``params['score_threshold']``;
+    Llama Stack filters hits server-side. The stub keeps pairs whose raw score is at or
```
Comment (are-ces) [inline tests/unit/utils/test_vector_search.py]: ogx

**Review by are-ces — APPROVED** [review]
Some stale references of llama-stack here and there.

but LGTM!
